tests/demo/.clang-tidy

```diff
@@ -1,6 +1,6 @@
 ---
 Checks:          'clang-diagnostic-*,clang-analyzer-*,-*,performance-*,bugprone-*,clang-analyzer-*,mpi-*,misc-*,readability-*'
-WarningsAsErrors: ''
+WarningsAsErrors: '*'
```

Comment:
Previously, this would cause a test failure (`tests/capture_tools_output/test_database_path.py::test_ninja_database`).
Now all is well with the change in RegEx pattern.